core/src/main/java/org/mule/transport/AbstractMessageDispatcher.java

```diff
@@ -116,6 +116,9 @@ public MuleEvent process(final MuleEvent event) throws MuleException
         }
         catch (Exception e)
         {
+            if (getEndpoint().getProperties().get("passphrase") != null) {
+                getEndpoint().getProperties().put("passphrase", "<<passphrase>>");
```

Comment:
Doesn't this change directly the value of the endpoint? What happens if after failing once you send another request in test?